lib/ddtrace/contrib/httprb/datadog_wrap.rb

```diff
@@ -0,0 +1,109 @@
+
+require 'uri'
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/http'
+require 'ddtrace/ext/net'
+require 'ddtrace/ext/distributed'
+require 'ddtrace/contrib/analytics'
+require 'ddtrace/propagation/http_propagator'
+
+require 'http'
+
+module Datadog
+  module Contrib
+    module Httprb
+      class DatadogWrap < ::HTTP::Feature
+
+        def initialize(*)
+          super
+        end
+
+        def wrap_request(request)
+          begin
+            return super(request) unless tracer_enabled?
+
+            datadog_span = datadog_configuration[:tracer].trace(
+              Ext::SPAN_REQUEST,
+              service: datadog_configuration[:service_name],
+              span_type: Datadog::Ext::HTTP::TYPE_OUTBOUND
+            )
+            
+            if request && request.headers
+              Datadog::HTTPPropagator.inject!(datadog_span.context, request.headers)
+            end
+
+            Contrib::Analytics.set_sample_rate(datadog_span, analytics_sample_rate) if analytics_enabled?
+
+            http_method = request.try(:verb).try(:to_s).try(:upcase)
```

Comment:
Could you double-check that `try` is available when running this code from an app?
Running it in our test environment might include additional dependencies that provide this method, so it wouldn't be an accurate test.

Reply:
ah, yup, will do, just force of habit